patches/woocommerce-order-status-manager.0001.fix.cache-invalidation.patch

```diff
@@ -0,0 +1,41 @@
+From 54447cc8207545df3635b86a5c7aeb98db100a0e Mon Sep 17 00:00:00 2001
+From: Nabi <artsanaat@gmail.com>
+Date: Wed, 30 Nov 2022 11:05:12 +0300
+Subject: [PATCH] Fixed object cache in woocommerce-order-status-manager
+ plugin.
+
+---
+ src/wc-order-status-manager-functions.php | 15 ++++++++++++++-
+ 1 file changed, 14 insertions(+), 1 deletion(-)
+
+diff --git a/src/wc-order-status-manager-functions.php b/src/wc-order-status-manager-functions.php
+index 56627038..cec2a840 100644
+--- a/src/wc-order-status-manager-functions.php
++++ b/src/wc-order-status-manager-functions.php
+@@ -48,8 +48,21 @@ function wc_order_status_manager_get_order_status_posts( $args = array() ) {
+ 		$posts = get_posts( wp_parse_args( $args, $defaults ) );
+ 
+ 		// expire cache after 1 second to avoid potential issues with persistent caching
+-		wp_cache_set( 'wc_order_status_manager_order_status_posts', $posts, null, 1 );
++		wp_cache_set( 'wc_order_status_manager_order_status_posts', $posts);
+ 	}
+ 
+ 	return $posts;
+ }
++
++/**
++ * Invalidates the object cache in Woocommerce order status manager.
++ *
++ * @implements save_post_wc_order_status
++ */
++function save_post_wc_order_status() {
++    wp_cache_get('wc_order_status_manager_order_status_posts', '', FALSE, $found);
++    if ($found) {
++        wp_cache_delete('wc_order_status_manager_order_status_posts');
++    }
++}
++add_action('save_post_wc_order_status', 'save_post_wc_order_status');
+\ No newline at end of file
```

Comment:
Please add a newline at the end of the file. You might have to change your editor preferences to not remove it automatically.